⚡️(vault) ask for vaultpass only when required

jmaupetit opened the pull request:
## Purpose

The vault password should only be required when creating secrets. Current configuration forced us to decrypt and load vaults variables whatever the playbook we were running.

## Proposal

- [x] remove `XXX_VAULT.*` invocation in objects that are not secrets
- [x] load application vault only when required (during secrets creation)

jmaupetit committed:
⚡️(vault) ask for vaultpass only when required

The vaultpass should only be required when creating secrets. Current
configuration forced us to decrypt and load vaults variables whatever
the playbook we were running.

sampaccoud commented:
This is awesome :muscle:

sampaccoud commented on apps/edxapp/vars/all/main.yml:
```diff
@@ -15,6 +15,8 @@ edxapp_sql_dump_url: "https://gist.githubusercontent.com/jmaupetit/76fe7db4a8314
 
 # i18n
+# Should we download and compile latest translations?
+edxapp_i18n_should_update: false
```
Maybe start with `should` then the verb?
```suggestion
edxapp_should_update_18n: false
```

jmaupetit replied:
Ah ah. I knew this was a bad name (talked about it with @lunika), but I kind like namespacing vars as `{{ application.name}}_{{ scope }}_{{ purpose }}`. But I can ignore this implicit rule and update this variable name. WDYT?

sampaccoud replied:
I would say namespacing with application name is fine. Scope may be too much since we have only a few settings?

jmaupetit replied:
Ok then, I'll do this.

sampaccoud commented on apps/marsha/templates/app/dc.yml.j2:
```diff
@@ -61,19 +61,15 @@ spec:
         - name: marsha-cloudfront-private-key-secret
           secret:
             secretName: "{{ marsha_cloudfront_private_key_secret_name }}"
-{% endif %}
```
just curious: do you remove it because in our case we always use it or because it works even if it is not set?

jmaupetit replied:
It is required, so I expect the secrets creation to fail if the vault entry does not exists.

sampaccoud replied:
actually in Marsha it's not required if you don't activate URL signing in CloudFront. I'm not sure though whether we fully planned this possibility in Arnold.

jmaupetit replied:
@lunika did. Anyway, remove request signature is not something we would recommend right?

lunika replied:
No and by default you cannot change this setting in our terraform so cloudfront will be configured to restrict viewer access. You have to change this setting in the cloudwatch interface.

sampaccoud replied:
so it's a half baked thing in Marsha because I put some Django settings to deactivate it :disappointed:. The flag in cloudfront would be easy to add. Should we create an issue for that?

jmaupetit replied:
Yes! Both here and in Marha's repository.

sampaccoud replied:
ok let me do it.

jmaupetit replied:
I think it's only required for Marsha, I've rolled back this feature in Arnold.

jmaupetit committed:
🎨(logs) switch to native variable debugging

The YAML renderer combined with msg debug is no longer needed since we
switched back to native ansible logs rendering.

jmaupetit commented on apps/marsha/templates/app/secret_cloudfront_private_key.yml.j2:
```diff
@@ -1,4 +1,3 @@
-{% if MARSHA_VAULT.DJANGO_CLOUDFRONT_PRIVATE_KEY is defined %}
```
Maybe I should rollback this to make `DJANGO_CLOUDFRONT_PRIVATE_KEY` optional in Marsha's vault? Ping @lunika
